src/System.Reflection/tests/MethodInfo/MethodInfo_InvokeTests.cs

```diff
@@ -259,13 +393,126 @@ public static MethodInfo getMethod(Type t, string method)
             {
                 if (alldefinedMethods.Current.Name.Equals(method))
                 {
-                    //found method
+                    // Found the method
                     mi = alldefinedMethods.Current;
                     break;
                 }
             }
             return mi;
         }
+
+        private interface IDefaultParameters
+        {
+            string InterfaceMethod(int p1 = 1, string p2 = "test", decimal p3 = 3.14m);
+        }
+
+        private class DefaultParameters : IDefaultParameters
+        {
+            public int Integer(int param = 1)
+            {
+                return param;
+            }
+
+            public string AllPrimitives(
+                bool boolean = true,
+                string str = "test",
+                char character = 'c',
+                byte unsignedbyte = 2,
+                sbyte signedbyte = -1,
+                Int16 int16 = -3,
+                UInt16 uint16 = 4,
+                Int32 int32 = -5,
+                UInt32 uint32 = 6,
+                Int64 int64 = -7,
+                UInt64 uint64 = 8,
+                Single single = (Single)9.1,
+                Double dbl = 11.12)
+            {
+                StringBuilder builder = new StringBuilder();
+                builder.Append(boolean.ToString() + ", ");
+                builder.Append(str + ", ");
+                builder.Append(character.ToString() + ", ");
+                builder.Append(unsignedbyte.ToString() + ", ");
+                builder.Append(signedbyte.ToString() + ", ");
+                builder.Append(int16.ToString() + ", ");
+                builder.Append(uint16.ToString() + ", ");
+                builder.Append(int32.ToString() + ", ");
+                builder.Append(uint32.ToString() + ", ");
+                builder.Append(int64.ToString() + ", ");
+                builder.Append(uint64.ToString() + ", ");
+                builder.Append(single.ToString() + ", ");
+                builder.Append(dbl);
+                return builder.ToString();
+            }
+
+            public string String(string parameter = "test")
```

Comment:
Nit: If you wanted, functions like this could be simplified a bit with expression-bodied members, e.g.
```C#
public string String(string parameter = "test") => parameter;
```